lib/libzrepl/data_conn.c

```diff
@@ -348,6 +356,8 @@ uzfs_zvol_worker(void *arg)
 
 		case ZVOL_OPCODE_SYNC:
 			uzfs_flush_data(zinfo->main_zv);
```

Comment:
also we should flush to main_zv if (ZVOL_IS_REBUILDING_AFS(zinfo->main_zv) || ZVOL_IS_HEALTHY(zinfo->main_zv).

Reply:
done